### ISSUE TITLE
Library is not consumable from CJS context in TypeScript application

### ISSUE THREAD

**Issue by IdkMan2Usertive** [i18next/i18next-fs-backend#35]
## 🐛 Bug Report

Library cannot be consumed from the CJS context when using TypeScript as of TS1479 error:

```
TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("i18next-fs-backend")' call instead.   To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.
```

## To Reproduce

https://codesandbox.io/s/i18next-fs-backend-issue-s4qlzc?file=/src/index.ts

## Expected behavior

We should be able to import this library both from CJS context and ESM context.

## Your Environment

- *runtime version*: node v18
- *i18next version*: ^22.4.9
- *os*: Mac

## Solution  🔥 

In order to solve the issue you have to provide two different types declaration files, for CJS and for ESM.

I played around a bit with the package.json configuration and found out that this one is actually working, allowing me to consume this library, not matter the context (CJS / ESM).

Although "./cjs/index.d.ts" and "./esm/index.js" is the same file.

```json
{
  "type": "module",
  "main": "./cjs/index.js",
  "module": "./esm/index.js",
  "types": {
    "require": "./cjs/index.d.ts",
    "import": "./esm/index.d.ts"
  },
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "types": {
        "require": "./cjs/index.d.ts",
        "import": "./esm/index.d.ts"
      },
      "require": "./cjs/index.js",
      "default": "./esm/index.js"
    },
    "./cjs": {
      "types": "./cjs/index.d.ts",
      "default": "./cjs/index.js"
    },
    "./esm": {
      "types": "./esm/index.d.ts",
      "default": "./esm/index.js"
    }
  },
}
```

## Related topics 🔥 

https://github.com/microsoft/TypeScript/issues/49299
https://github.com/Swatinem/rollup-plugin-dts/issues/210

## The only one workaround for now

```
// @ts-ignore
```

😢

**Comment by adrai** [comment]
sorry, but this can't be the official way TypeScript wants us package maintainers to handle it...
//cc: @pedrodurek

**Comment by e6c31d** [comment]
PR to fix this issue: #36

**Comment by adrai** [comment]
should be fixed with v2.1.2
